### PR TITLE
Adds config option to blacklist Exploration Crew from antagonist & antagonist objectives.

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request
Title.

## Why It's Good For The Game
Exploration is currently basically required for the station to function properly, especially in the funds department. They hardly interact with the actual core game outside of dumping massive amounts of money onto the station, and if people want to go off and play a co-op game instead of SS13, they shouldn't really have the chance to be given antag to murder their co-op buddies who will never have the chance to play again for that round.

## Changelog
:cl:
config: Added configuration option to blacklist Exploration Crew from antagonist status and objectives.
tweak: Exploration Crew can no longer be antagonists or their target.
/:cl:

### CODE REVIEW

File: code/datums/helper_datums/getrev.dm [hunk 1]
```diff
@@ -89,6 +89,7 @@
 	msg += "Protect Authority Roles From Traitor: [CONFIG_GET(flag/protect_roles_from_antagonist)]"
 	msg += "Protect Assistant Role From Traitor: [CONFIG_GET(flag/protect_assistant_from_antagonist)]"
 	msg += "Protect Command Roles From Traitor: [CONFIG_GET(flag/protect_heads_from_antagonist)]"
+	msg += "Protect Exploration From Traitor: [CONFIG_GET(flag/protect_exploration_from_antagonist)]"
```
Comment (KubeRoot) [inline code/datums/helper_datums/getrev.dm]: Do I understand correctly that this is the only one of these that protects from antagonist *targets* too? If yes, maybe it should be noted in the message?

File: config/game_options.txt [hunk 2]
```diff
@@ -263,6 +263,9 @@ PROTECT_ROLES_FROM_ANTAGONIST
 ## Uncomment to prohibit head roles from becoming most antagonists.
 PROTECT_HEADS_FROM_ANTAGONIST
 
+## Unciomment to prohibit exploration roles from becoming antagonists or their targets.
```
Comment (KubeRoot) [inline config/game_options.txt]: Typo in `Uncomment`

File: code/datums/helper_datums/getrev.dm (hunk 1 above)
Comment (Kapu1178) [reply code/datums/helper_datums/getrev.dm]: Its in the config option as that.

File: config/game_options.txt (hunk 2 above)
Comment (Kapu1178) [reply config/game_options.txt]: Thanks

File: code/datums/helper_datums/getrev.dm (hunk 1 above)
Comment (Kapu1178) [reply code/datums/helper_datums/getrev.dm]: updated anyway cuz why not

**Comment by Sinestia** [comment]
This is a pretty sad change to be having to make lmfao 


But, probably the most direct way to resolve some of the problems with the job. 

Edit: I'm not really sure if I agree with your reasoning though, I think the same reasoning would easily justify a mode like yog's that completely disables being an antag target for anybody. The main issue with antag explorers is that it's unbalanced and that with the current tone of the game you can't expect people to at least make it interesting.

**Comment by DerpiestBruh** [comment]
I feel like this is going to be metagamed as hell.

**Comment by MNarath1** [comment]
This is litterally the same issue miners already had for years:
1.Important for the station yes
2.Mostly only two people yes
3.Bane of antags as targets that either you will not need to do anything cause they die or they will come back packed with weapons also yes

Also i don't know how often you see people getting murdered by their partner i had antags as partner like 2 times the possibility you are their target is extremly low and with current server rules they aren't even allowed to just outright kill you the only way you probably get problems is if you partner is like a changeling also antags(at least traitor) have certainly a way to deal with targets that are exploring litterally a special shuttle they can buy with their uplink. Also there are lots of other possible ways to get back into the round beside getting your body back you gonna have to accept that there is the possibility that you just won't be able to play your current character anymore.

**Comment by MNarath1** [comment]
I really don't think we need another "antag proof" job security is already bad enough but at least they can be justified somewhat this trough no

**Comment by Kapu1178** [comment]
> This is litterally the same issue miners already had for years: 1.Important for the station yes 2.Mostly only two people yes 3.Bane of antags as targets that either you will not need to do anything cause they die or they will come back packed with weapons also yes
> 
> Also i don't know how often you see people getting murdered by their partner i had antags as partner like 2 times the possibility you are their target is extremly low and with current server rules they aren't even allowed to just outright kill you the only way you probably get problems is if you partner is like a changeling also antags(at least traitor) have certainly a way to deal with targets that are exploring litterally a special shuttle they can buy with their uplink. Also there are lots of other possible ways to get back into the round beside getting your body back you gonna have to accept that there is the possibility that you just won't be able to play your current character anymore.

Miners are much easier to replace than Exploration Crews.

**Comment by EvilDragonfiend** [comment]
I think spawning them as an antag is not bad when there are 3 exploration crews existing in a shift, and only one of them should be an antag. I won't say they need a security-level non-antag guarantee, but they should have a less chance at least.
if we shouldn't do this, then we should lift our station dependency on them.

**Comment by OceanFish1** [comment]
Kinda agree, I was getting expo coworker as my antag target too often

At least instead of backstab I was making fair duels

**Comment by MNarath1** [comment]
> > This is litterally the same issue miners already had for years: 1.Important for the station yes 2.Mostly only two people yes 3.Bane of antags as targets that either you will not need to do anything cause they die or they will come back packed with weapons also yes
> 
> > 
> 
> > Also i don't know how often you see people getting murdered by their partner i had antags as partner like 2 times the possibility you are their target is extremly low and with current server rules they aren't even allowed to just outright kill you the only way you probably get problems is if you partner is like a changeling also antags(at least traitor) have certainly a way to deal with targets that are exploring litterally a special shuttle they can buy with their uplink. Also there are lots of other possible ways to get back into the round beside getting your body back you gonna have to accept that there is the possibility that you just won't be able to play your current character anymore.
> 
> 
> 
> Miners are much easier to replace than Exploration Crews.

Not really the only real unique item exploration crew has is the gun thats basically a weaker energy gun you can't fire on the station(honestly super useless especially cause exploration crew finds guns during their missions anyways so why locking the starter gun to anywhere but the station) and maybe the hardsuit but both can be replaced and i would argue that you don't need the gun either and can probably do with a kinetic accelerator too (literally like miners) especially if you are already decompressing the area too there are other unique items like the disk locator but those are roundstart spawn only so they litterally cannot be replaced and i would argue its not really a necessity as you can find disks very easily without it too.

**Comment by DatBoiTim** [comment]
> Kinda agree, I was getting expo coworker as my antag target too often

The disk locater is actually unobtainable if you don't spawn with it.

**Comment by ike709** [comment]
Off-station jobs shouldn't get rev but otherwise I disagree with this PR. 🔑

**Comment by PowerfulBacon** [comment]
This is a bad PR and I don't think the reasoning supports it. At this point then why shouldn't miners be protected from antagonists and just because people want to go off and play a coop game doesn't mean we should allow them to be completely isolated from the station. This makes exploration even more of a bubble. And also there are a lot of jobs that are required for the station to function, engineers, scientists, cargo, mining etc. but they aren't protected from antagonists.

**Pull request closed without merging**